SalesforceSDK/Salesforce.SDK.SmartSync/Manager/SyncManager.cs

```diff
@@ -81,17 +83,25 @@ private SyncManager(Account account, string communityId)
         /// <returns></returns>
         public static SyncManager GetInstance(Account account, string communityId = null)
         {
+            return GetInstance(account, communityId, null);
+        }
+
+        public static SyncManager GetInstance(Account account, string communityId, SmartStore.Store.SmartStore smartStore)
+        {
             if (account == null)
             {
                 account = AccountManager.GetAccount();
             }
-            if (account == null)
+
+            if (smartStore == null)
             {
-                return null;
+                smartStore = SmartStore.Store.SmartStore.GetSmartStore(account);
             }
+
             string uniqueId = Constants.GenerateAccountCommunityId(account, communityId);
```

Comment:
uniqueId should be computed from smartstore path and user account:
For Android we do:
        String uniqueId = (account != null ? account.getUserId() : "") + ":"
                    + smartStore.getDatabase().getPath();